model/attention.py

```diff
@@ -0,0 +1,95 @@
+import torch
+import torch.nn as nn
+import math
+
+
+def dot_product_attention(q, k, v):
+    """
+    Function that computes dot product attention between
+    query, key, value vectors
+    """
+    # TODO, ADD MASKING
+    c = q.shape[-1]
+    attn = torch.matmul(q, k.transpose(-1, -2)) / math.sqrt(c)
```

Comment:
Does `matmul` work when inputs have more than 3 dimensions (e.g. B x N x HEADS x N x E)? Not sure if we need to merge some dimensions.